tiling.js

```diff
@@ -1476,19 +1483,46 @@ class Spaces extends Map {
         }
     }
 
-    selectSpace(direction, move, transition) {
+    _getSpacesStack(use_mru) {
+        let spaces = [];
+
+        // get current workspace and current workspaceindex
+        let currentSpace = this.spaceOf(workspaceManager.get_active_workspace());
+        let currentSpace_index = currentSpace.workspace.index();
+
+        // use the mru stack to get array of spaces
+        if (use_mru === true) {
+            spaces = [...this.stack];
+            // go over each monitor and collect spaces
+            this.monitors.forEach(space => spaces.splice(spaces.indexOf(space), 1));
+            spaces = [currentSpace, ...spaces];
+        }
+        // get a sequential array of spaces
+        else {
+            let workspaces_count = workspaceManager.get_n_workspaces();
+            for (let i = 0; i < workspaces_count; i++) {
+                let workspace = workspaceManager.get_workspace_by_index(i);
+                spaces.push(this.get(workspace));
```

Comment:
Here we might want to do something like this to limit the spaces to the ones that are on the current monitor:
```javascript
let space = this.get(workspace);
if (space.monitor === currentSpace.monitor)
  spaces.push(space)
```